features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.dashboard/dashboard.js

```diff
@@ -28,6 +28,11 @@ function onRequest(context) {
     var deviceModule = require("/app/modules/device.js").deviceModule;
     var policyModule = require("/app/modules/policy.js").policyModule;
 
+    var permissions = userModule.getUIPermissions();
+    if (!permissions.VIEW_DASHBOARD) {
+        response.sendRedirect(constants.WEB_APP_CONTEXT + "/devices");
+    }
+
     context.device_count = deviceModule.getOwnDevicesCount();
     context.user_count = userModule.getUsers()["content"].length;
     context.policy_count = policyModule.getAllPolicies()["content"].length;
```

Comment:
Fixed in commit https://github.com/wso2/carbon-device-mgt/commit/e537e2d841ca3c7946cd34e9da96f261e0ac94e1